packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -0,0 +1,380 @@
+  /// value is true, and to false if value is null (i.e. it cycles through false
+  /// => true => null => false when tapped).
+  ///
+  /// If tristate is false (the default), [value] must not be null.
+  final bool tristate;
+
+  /// The color for the checkbox's border shadow when it has the input focus.
+  ///
+  /// If null, then a paler form of the [activeColor] will be used.
+  final Color? focusColor;
+
+  /// {@macro flutter.widgets.Focus.focusNode}
+  final FocusNode? focusNode;
+
+  /// {@macro flutter.widgets.Focus.autofocus}
+  final bool autofocus;
+
+  /// The color and width of the checkbox's border.
+  ///
+  /// If this property is null, then the side will be width 1.
+  final BorderSide? side;
+
+  /// The shape of the checkbox.
+  ///
+  /// If this property is null then the shape will be a [RoundedRectangleBorder]
+  /// with a circular corner radius of 4.0.
+  final OutlinedBorder? shape;
+
+  /// The width of a checkbox widget.
+  static const double width = 18.0;
+
+  @override
+  State<CupertinoCheckbox> createState() => _CupertinoCheckboxState();
+}
+
+class _CupertinoCheckboxState extends State<CupertinoCheckbox> with TickerProviderStateMixin, ToggleableStateMixin {
+  final _CheckboxPainter _painter = _CheckboxPainter();
+  bool? _previousValue;
+
+  bool focused = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _previousValue = widget.value;
+  }
+
+  @override
+  void didUpdateWidget(CupertinoCheckbox oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.value != widget.value) {
+      _previousValue = oldWidget.value;
+    }
+  }
+
+  @override
+  void dispose() {
+    _painter.dispose();
+    super.dispose();
+  }
+
+  @override
+  ValueChanged<bool?>? get onChanged => widget.onChanged;
+
+  @override
+  bool get tristate => widget.tristate;
+
+  @override
+  bool? get value => widget.value;
+
+  void onFocusChange(bool value) {
+    if (focused != value) {
+      focused = value;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+
+    final Color effectiveActiveColor = widget.activeColor
+      ?? CupertinoColors.activeBlue;
+    final Color? inactiveColor = widget.inactiveColor;
+    final Color effectiveInactiveColor = inactiveColor
+      ?? CupertinoColors.inactiveGray;
+
+    final Color effectiveFocusOverlayColor = widget.focusColor
+      ?? HSLColor
+          .fromColor(effectiveActiveColor.withOpacity(0.80))
+          .withLightness(0.69).withSaturation(0.835)
+          .toColor();
+
+    final Color effectiveCheckColor = widget.checkColor
+      ?? CupertinoColors.white;
+
+    return Semantics(
+      checked: widget.value ?? false,
+      mixed: widget.tristate ? widget.value == null : null,
+      child: buildToggleable(
+        focusNode: widget.focusNode,
+        autofocus: widget.autofocus,
+        onFocusChange: onFocusChange,
+        size: const Size(48.0, 48.0),
```

Comment:
Should this be slightly smaller?

```suggestion
        size: const Size.square(kMinInteractiveDimensionCupertino),
```
Which would make it 44x44. 48 is the Material min interactive dimension for tap targets.

Reply:
I was thinking the size should be about the same as the Material checkbox, so that they would take up about the same space for the eventual adaptive widget?

Reply:
Of at least be user configurable

Reply:
Adaptive widgets don't need to be the same size, and I think it's probably better to have it conform to the standard interactive size for Cupertino, since that should fit better with other Cupertino widgets. It's not much different anyhow.

Reply:
I don't think we need to make the size configurable. If we were going to make a more general purpose toggle widget, then yes, but these are supposed to look like macOS checkboxes, and they all have the same size. The Material checkboxes are fixed size.